### DISCUSSION TITLE
Problem with hitting lower cut off voltage in MSMR modelling of full cells

### DISCUSSION THREAD

**Discussion by akhilmabraham** [pybamm-team/PyBaMM#4368] [Q&A] [answered]
Hello, 

I am trying to simulate a full cell using MSMR model. To avoid any confusion, I am calling X, U and W as the MSMR parameters. The MSMR parameters for anode (X, U and W for 6 galleries) was retrieved after fitting the half cell curve to a C/50 lithiation cycle. The default MSMR parameters in pybamm were used for the cathode open circuit potential (NMC 622 de/lithiated at C/100). I would very much appreciate help with the following @TomTranter @valentinsulzer @rtimms :
**a.** Not able to run full cell simulation between the voltage window 2. 8 - 4.2 V. However when I reduce the LCV to 3.25 the simulation runs fine. I am guessing this has to do with OCP balancing. How do I do that? Can i do this by adjusting the maximum concentration of negative and positive electrode? 
**b.** If the cathode half cell is cycled at a different rate (C/100) than the anode for fetching MSMR parameters will this have any influence on full cell modelling
**c.** Sometimes I come across the error - "At t = 294.091 and h = 9.58025e-06, the corrector convergence failed repeatedly or with |h| = hmin" -  I understand that the solver is facing a convergence issue. I can potentially rectify by specifying low dt_max or time period. What does t and h mean?

Find below the output from simulation runs and attached the code. Thanks a ton in advance

    Output for full cell simulation between 3.25 - 4.2 V
![image](https://github.com/user-attachments/assets/20feae0f-8cc4-4fc6-bbda-15a84abf275f)

Output for full cell simulation between 2.8 - 4.2 V
![image](https://github.com/user-attachments/assets/3eba0aeb-0337-4ac3-8a9d-183fe16a25d0)

[Full cell simulation LCV - 2.8 V.pdf](https://github.com/user-attachments/files/16698564/Full.cell.simulation.LCV.-.2.8.V.pdf)

**Comment by valentinsulzer** [comment]
a. To do the full cell balancing you need to adjust maximum concentration as well as initial concentration in both electrodes
b. This shouldn't make too much difference
c. We are working on a new solver (IDAKLU) which should not have these errors. It will be fully released in v24.9

**Reply by akhilmabraham** [reply]
Thanks so much for your response @valentinsulzer.  
**a.** That's good to know. I have been playing around with max concentrations in the above script. Still facing converging issue at lower voltage, especially where the region where potential is dropping. Do you think using DFNmodel, passing MSMR model options will be a little more stable with CasadiSolver. For example:
 _model = pybamm.lithium_ion.DFN({"number of MSMR reactions": ('6', '4'),
                                      "intercalation kinetics": ("MSMR", "MSMR"),
                                      "particle": ("MSMR", "MSMR"),
                                      "open-circuit potential": ("MSMR", "MSMR"),
                                        })_. This will also allow me to pass initial concentration. I am not sure, employing MSMR model and updating initial concentration does anything different on the simulation output as it uses initial voltage of negative and positive electrodes to adjust the endpoints of anode and cathode half cells. 
**c.** Waiting for the new release! Is there a challenge running IDAKLU solver in version 24.5? IF there are things I should be wary about with current implementaion about IDAKLU that will be good to know. I will try IDAKLU anyways.

**Reply by valentinsulzer** [reply, marked as answer]
a. Ideally, when you are fitting the initial/max concentrations for the MSMR model, you would directly calculate the OCV vs capacity curve (using the MSMR model analytical expressions) instead of solving the full DFN model. Doing it this way you can get a fit in just a few seconds. It's a little bit complicated to set up though and this is what the Ionworks software does for you.
c. The IDAKLU solver in version 24.5 should work well already, but the version in v24.9 will be faster

**Reply by akhilmabraham** [reply]
Thanks @valentinsulzer. Adjusting the initial/max concentration worked for me with IDAKLU solver. The solver is not crashing at lower cut off voltages atleast for now for simulated cells. Computation is so much faster then Casadi.

**Reply by valentinsulzer** [reply]
Ok great

**Reply by vipulramk** [reply]
Hey just checking does the initial concentration set the SOC at t=0 of the simulation or is this parameter used to set the stochiometric limits
